### PR TITLE
XW-3671 | Move Maps buttons to new Page Header

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/XW-3671

@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php [hunk 1]
```diff
@@ -104,9 +104,21 @@ public static function onBeforePageDisplay( $out ) {
 	 * @return bool
 	 */
 	public static function onAfterPageHeaderButtons( \Title $title, array &$buttons ): bool {
```
Comment (WikiaTech) [inline extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php]: MAJOR Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 10 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaMapsHooks.class.php#L26](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php#L26): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [WikiaMapsHooks.class.php#L43](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php#L43): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WikiaMapsHooks.class.php#L55](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php#L55): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WikiaMapsHooks.class.php#L75](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php#L75): Remove the unused function parameter "$jsStaticPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [WikiaMapsSpecialController.class.php#L214](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/controllers/WikiaMapsSpecialController.class.php#L214): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MINOR][MINOR] [WikiaMapsSpecialController.class.php#L311](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/controllers/WikiaMapsSpecialController.class.php#L311): Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. ![MINOR][MINOR] [WikiaMapsSpecialController.class.php#L311](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/controllers/WikiaMapsSpecialController.class.php#L311): Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. ![MINOR][MINOR] [WikiaMapsSpecialController.class.php#L312](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/controllers/WikiaMapsSpecialController.class.php#L312): Remove the literal "true" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)
1. ![MINOR][MINOR] [WikiaMapsSpecialController.class.php#L312](https://github.com/Wikia/app/blob/b6c09a00d8b7e1a8175c727bf4bcedba22cb5502/extensions/wikia/WikiaMaps/controllers/WikiaMapsSpecialController.class.php#L312): Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaMapsHooks.class.php#L26](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php#L26): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [WikiaMapsHooks.class.php#L43](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php#L43): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WikiaMapsHooks.class.php#L55](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php#L55): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WikiaMapsHooks.class.php#L75](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php#L75): Remove the unused function parameter "$jsStaticPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [WikiaMapsSpecialController.class.php#L214](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/controllers/WikiaMapsSpecialController.class.php#L214): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WikiaMapsSpecialControllerTest.php#L186](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/tests/WikiaMapsSpecialControllerTest.php#L186): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WikiaMapsSpecialControllerTest.php#L202](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/tests/WikiaMapsSpecialControllerTest.php#L202): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WikiaMapsSpecialControllerTest.php#L392](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/tests/WikiaMapsSpecialControllerTest.php#L392): Add a "case default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
1. ![MAJOR][MAJOR] [WikiaMapsSpecialControllerTest.php#L396](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/tests/WikiaMapsSpecialControllerTest.php#L396): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [WikiaMapsSpecialControllerTest.php#L400](https://github.com/Wikia/app/blob/2f8fd0a08c7403b402f3711457626c7233b39f3a/extensions/wikia/WikiaMaps/tests/WikiaMapsSpecialControllerTest.php#L400): Reduce this "switch/case" number of lines from 12 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by KockaAdmiralac** [comment]
Is this pull request missing a ticket name in the title/description?

**Comment by rogatty** [comment]
@KockaAdmiralac not every PR has a related ticket, this is one of them. Thanks for keeping an eye :policeman:

**Review by hakubo — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php [hunk 2]
```diff
@@ -104,9 +107,25 @@ public static function onBeforePageDisplay( $out ) {
...
-		if ( $title->isSpecial( 'Maps' ) ) {
+		if ( self::isSpecialMapsPage() ) {
 			$label = wfMessage( 'wikia-interactive-maps-create-a-map' )->escaped();
 			$buttons[] = new Button( $label, '', '#', '', 'createMap' );
+
+			if ( self::isSingleMapPage() ) {
+				global $wgIntMapConfig;
+
+				$mapId = explode( '/', $title->getText() )[1]; // the page name (without ns prefix) is build like this: Maps/23414
```
Comment (WikiaTech) [inline extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Review by hakubo — APPROVED** [review]

File: extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php [hunk 3]
```diff
@@ -104,9 +107,21 @@ public static function onBeforePageDisplay( $out ) {
 	 * @return bool
 	 */
 	public static function onAfterPageHeaderButtons( \Title $title, array &$buttons ): bool {
```
Comment (WikiaTech) [inline extensions/wikia/WikiaMaps/WikiaMapsHooks.class.php:109]: MAJOR Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by hakubo — APPROVED** [review]
yup, ugly. but let's move on

**Review by rogatty — COMMENTED** [review]
Changes LGTM

**Pull request merged**